tests/test_parsers.py

```diff
@@ -79,23 +79,50 @@ def test_gelfparser_parse_partially_invalid_file(caplog):
                 # Another assumed valid gelf
                 '{"short_message": {"username": "This seems valid too."}}\n',
             ]
+        )        
+
+        with caplog.at_level(logging.DEBUG):
+            file.seek(0)
+            parser = GELFParser()
+            events = list(parser.parse(file))
+
+        assert(
+            (
+            "Input event '{ This is not valid json and raises json.decoder.JSONDecodeError\n' "
+            "is not a valid JSON string! It will be ignored."
+            )
+            in caplog.text
         )
+
+        assert(
+            "Input event '{}\n' doesn't comply with GELF format! It will be ignored."
+            in caplog.text
+        )
+
+        assert(
+            "Input event '[]\n' is not a valid JSON string! It will be ignored."
+            in caplog.text
+        )
+
         with caplog.at_level(logging.WARNING):
             file.seek(0)
             parser = GELFParser()
             events = list(parser.parse(file))
-            parser.parse(file)
 
-        file.seek(0)
-        assert (
+        assert(
+            (
             "Input event '{ This is not valid json and raises json.decoder.JSONDecodeError\n' "
             "is not a valid JSON string! It will be ignored."
-        ) in caplog.text
-        assert (
+            )
+            in caplog.text
+        )
+
+        assert(
             "Input event '{}\n' doesn't comply with GELF format! It will be ignored."
             in caplog.text
         )
-        assert (
+
+        assert(
             "Input event '[]\n' is not a valid JSON string! It will be ignored."
             in caplog.text
         )
```

Comment:
I would test the WARNING level first, then DEBUG.

Reply:
You're right, it makes sense